staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -484,7 +507,7 @@ func (s sortableStoreElements) Swap(i, j int) {
 // WaitUntilFreshAndList returns list of pointers to `storeElement` objects along
 // with their ResourceVersion and the name of the index, if any, that was used.
 func (w *watchCache) WaitUntilFreshAndList(ctx context.Context, resourceVersion uint64, matchValues []storage.MatchValue) ([]interface{}, uint64, string, error) {
-	err := w.waitUntilFreshAndBlock(ctx, resourceVersion)
+	err := w.waitUntilFreshAndBlock(ctx, resourceVersion, utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache))
```

Comment:
Readability: Wouldn't it be simpler to extract "requestBookmarkIfBehind" from the waitUntilFreshAndBlock?

We can write here something like:
```
if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) {
    w.maybeRequestBookmark(resourceVersion)
}

err := w.waitUntilFreshAndBlock(ctx, resourceVersion)
```

I get that there is a race between requesting bookmark and waiting, but I think it's a case anyway: maybe we have just requested bookmark already, maybe the event is on the way already?

I guess this allows to have two simpler functions instead of one with more complex semantics of `requestBookmarkIfBehind`?

Reply:
Done